plugins/network/NetClass.hpp

```diff
@@ -2,37 +2,110 @@
 #define __NETCLASS_HPP__
 
 #include <plugin.hpp>
+#include <memory>
+#include <string>
+#include <vector>
+
 #ifdef NETWORK_LOGGING
 #include <stdio.h>
 #endif
 
-class NetResourceList
+using string = std::wstring;
+
+class NetResource
 {
+public:
+	DWORD dwScope{};
+	DWORD dwType{};
+	DWORD dwDisplayType{};
+	DWORD dwUsage{};
+	std::unique_ptr<string> lpLocalName{};
+	std::unique_ptr<string> lpRemoteName{};
+	std::unique_ptr<string> lpComment{};
+	std::unique_ptr<string> lpProvider{};
+
 private:
-	NETRESOURCE* ResList;
-	unsigned int ResCount;
+	void copy(const NetResource& other)
+	{
+		dwScope = other.dwScope;
+		dwType = other.dwType;
+		dwDisplayType = other.dwDisplayType;
+		dwUsage = other.dwUsage;
+		lpLocalName = other.lpLocalName? std::make_unique<string>(*other.lpLocalName) : nullptr;
+		lpRemoteName = other.lpRemoteName? std::make_unique<string>(*other.lpRemoteName) : nullptr;
+		lpComment = other.lpComment? std::make_unique<string>(*other.lpComment) : nullptr;
+		lpProvider = other.lpProvider? std::make_unique<string>(*other.lpProvider) : nullptr;
+	}
 
 public:
-	NetResourceList();
-	~NetResourceList();
+	NetResource() = default;
+
+	NetResource(const NetResource& other)
+	{
+		copy(other);
```

Comment:
Instead of having a private copy function it can be delegated to the operator, e.g. `*this = other;`